source/reports/ml_reports/ROCCurve.py

```diff
@@ -0,0 +1,70 @@
+from source.reports.ReportOutput import ReportOutput
+from source.reports.ReportResult import ReportResult
+from source.reports.ml_reports.MLReport import MLReport
+from sklearn.metrics import roc_curve, auc
+
+import warnings
+import plotly.graph_objs as go
+import os
+import numpy as np
+
+
+class ROCCurve(MLReport):
+
+    def __init__(self, name: str = None):
+        super().__init__()
+        self.name = name
+
+    @classmethod
+    def build_object(cls, **kwargs):
+        name = kwargs["name"] if "name" in kwargs else "ROC_curve"
+        return ROCCurve(name=name)
+
+    def generate(self) -> ReportResult:
+        x = self.test_dataset.encoded_data
+        y_score = self.method.predict_proba(x, self.label)[self.label]
+        fpr, tpr, _ = roc_curve(x.labels[self.label], y_score[:, 0])
+        roc_auc = auc(fpr, tpr)
+
+        trace1 = go.Scatter(x=fpr, y=tpr,
+                            mode='lines',
+                            line=dict(color='darkorange', width=2),
+                            name=f"ROC curve (area = {roc_auc})")
+        trace2 = go.Scatter(x=[0, 1], y=[0, 1],
+                            mode='lines',
+                            line=dict(color='navy', width=2, dash='dash'),
+                            showlegend=False)
+        layout = go.Layout(title='Receiver operating characteristic example',
+                           xaxis=dict(title='False Positive Rate'),
+                           yaxis=dict(title='True Positive Rate'))
+
+        fig = go.Figure(data=[trace1, trace2], layout=layout)
+
+        os.makedirs(self.result_path)
```

Comment:
here we have a class called PathBuilder with method build_path() which can be used (it additionally checks if the directory already exists etc) :)